src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/ManifestReader.cs

```diff
@@ -343,14 +345,20 @@ private static RepositoryMetadata ReadRepository(XElement element)
             {
                 repository.Url = repositoryUrl.Value;
             }
-            if (string.IsNullOrEmpty(repository.Type) && string.IsNullOrEmpty(repository.Type))
+
+            if (!string.IsNullOrEmpty(repository.Type))
```

Comment:
what if type is empty but branch/commit is not?

Reply:
The easy answer is we allow any combination. But if nuget.org or another tool were to consume this, I don't know how it could do anything without knowing if the url is pointing to a local folder of files, a git repo, or a cvs repository. It feels like we should enforce *something* here.